crates/ruma-state-res/src/room_version.rs

```diff
@@ -95,6 +101,7 @@ impl RoomVersion {
         special_case_aliases_auth: false,
         strict_canonicaljson: true,
         limit_notifications_power_levels: true,
+        has_aliases: false,
```

Comment:
This is adding this new rule for V6 specifically, while above it says "m.room.aliases is removed for room versions > 6."? Shouldn't this be added to V7 then?

Reply:
Every version is just the previous version with some fields overridden. Since `V7` doesn't override `has_aliases`, it inherits `false` from `..Self::V6`.

Reply:
Yes, but were aliases specifically removed in v6, or were they removed *after* v6? because this bit removes them in v6.

Reply:
Now I see. Honestly I don't think it's right to introduce another vaariable here. `special_case_aliases_auth` seems like it fills the exact same rule.